enhancements/oc/cli_manager.md

```diff
@@ -0,0 +1,137 @@
+location.  Each component is responsible for serving their files and registering information regarding supplied binaries
+to a central official location.  The CLI manager will reach out to the central place and retrieve all the binaries from there.
+Similarly, [kubectl plugin manager](https://github.com/GoogleContainerTools/krew/) has an [index](https://github.com/kubernetes-sigs/krew-index)
+to maintain all known krew plugins. 
+
+## Motivation
+
+As more services are created on top of OpenShift, more CLIs are introduced to simplify interaction with these services.
+Some current examples are oc, kubectl, odo, istio, tekton, and knative.  It is difficult for users to discover what tools exist, 
+where to download them from and which version they should download.  We need to simplify as much as possible the interaction
+of services on OpenShift.  This is the motivation for the CLI manager.
+
+### Goals
+
+Each component wishing to provide customers with their binaries will build and publish them via an official channel to an
+index location from which the CLI manager will obtain the files.
+
+The CLI manager will:
+- List available CLI plugins - their latest version and/or the installed version
+- Install appropriate versions of a chosen CLI for a given cluster
+- Update installed version of a given CLI if/when updates become available
+
+CLIs to manage:
+- **openshift-install** OpenShift Installer 
+- **kubectl** Kubernetes
+- **oc** OpenShift
+- **odo** OpenShift Developer
+- **kn** Knative
+- **tkn** Tekton
+
+### Non-Goals
+
+CLI manager will not build or serve the binaries.  It will know where to find them.
+
+## Proposal
+
+
+### User Stories [optional]
+
+#### Story 1
+
+As a user, I want a CLI manager for various CLIs available for OpenShift and related services so that I can discover, install and upgrade them
+
+##### example:
+```
+$ openshift-cli addons list
+NAME                 DESC                        LATEST       INSTALLED
+-----                -----                       -----        -----
+openshift-installer  OpenShift Installer CLI     4.4          4.3
+kubectl              Kubernetes CLI              1.15         1.13
+oc                   OpenShift CLI               4.4          4.3
+odo                  OpenShift Developer CLI     1.0          Not Installed
+kn                   Knative CLI                 1.0          Not Installed
+tkn                  Tekton CLI                  1.0          Not Installed
+
+
+$ openshift-cli addons install odo
+$ openshift-cli addons install kn
+$ openshift-cli addons update oc
+$ openshift-cli addons remove kn
+```
+## Design Details
+
+### Test Plan
+
+### Graduation Criteria
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned
+[maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+##### Tech Preview -> GA 
+
+### Upgrade / Downgrade Strategy
+
+### Version Skew Strategy
```

Comment:
Let's not cover the version skew, I'd just focus on the mechanics. The rest will be extensions we can work once we have the building blocks in place.

Reply:
removed